### PR TITLE
IQE-593: enhance testing

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

File: testing/conftest.py [hunk 1]
```diff
@@ -5,58 +5,78 @@
...
+        yield f"http://{forced_host}:4444/wd/hub"
 
 
 @pytest.fixture(scope="session")
-def wait_for_selenium(selenium_host):
-    @wait_for_decorator(timeout=10, handle_exception=True)
-    def make_request():
-        urlopen(f"http://{selenium_host}:4444/wd/hub")
+def wait_for_selenium(selenium_url):
+    wait_for(lambda: urlopen(selenium_url), timeout=10, handle_exceptions=True)
```
Comment (quarckster) [inline testing/conftest.py:72]: That's a root cause of why the tests fail:

```python
wait_for(lambda: urlopen(selenium_url), timeout=10, handle_exception=True)
```

I wanted to make in this as well but I guess made the same mistake. I mixed up `handle_exception` with `handle_exceptions`.

File: .github/workflows/workflow.yaml [hunk 2]
```diff
@@ -34,11 +34,13 @@ jobs:
...
           pip install pytest pytest-cov codecov pytest-xdist
           python setup.py install
+      - name: fetch podman image
+        run: sudo podman pull quay.io/redhatqe/selenium-standalone:latest
       - name: Test with pytest
         env:
           BROWSER: ${{ matrix.browser }}
         run: |
-          pytest -v -n 5 --no-cov-on-fail --cov=widgetastic_patternfly4 --cov-report=xml:/tmp/coverage.xml
+          pytest -v -s --no-cov-on-fail --cov=widgetastic_patternfly4 --cov-report=xml:/tmp/coverage.xml
```
Comment (quarckster) [inline .github/workflows/workflow.yaml:43]: It can be reverted back to parallel running.

File: .github/workflows/workflow.yaml [hunk 3]
```diff
@@ -34,11 +34,13 @@ jobs:
           pip install -U setuptools pip
           pip install pytest pytest-cov codecov pytest-xdist
           python setup.py install
+      - name: fetch podman image
+        run: sudo podman pull quay.io/redhatqe/selenium-standalone:latest
```
Comment (quarckster) [inline .github/workflows/workflow.yaml:38]: This one is not necessary.

File: testing/conftest.py [hunk 4]
```diff
@@ -5,58 +5,78 @@
...
-    subprocess.run(["sudo", "podman", "kill", container_id])
+def selenium_url(pytestconfig, worker_id):
+    forced_host = pytestconfig.getoption("--force-host")
+
+    if forced_host is None:
+        oktet = 1 if worker_id == "master" else int(worker_id.lstrip("gw")) + 1
+        host = f"127.0.0.{oktet}"
+        ps = subprocess.run(
+            [
+                "sudo",
```
Comment (quarckster) [inline testing/conftest.py:46]: According to https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md Actions nodes should have `podman 2.0.4` that has fixed that issue https://github.com/containers/podman/issues/7016. You can try to remove it.

File: testing/conftest.py [hunk 5]
```diff
@@ -5,58 +5,78 @@
...
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--browser-name",
+        help="name of the browser",
+        choices=("firefox", "chrome"),
+        default="firefox",
+    )
+
+    parser.addoption("--force-host", default=None, help="foce a selenium hostname")
```
Comment (quarckster) [inline testing/conftest.py:29]: typo:
`help="force a selenium hostname"`

File: testing/conftest.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply testing/conftest.py:72]: Yikes thanks

File: .github/workflows/workflow.yaml (hunk 3 above)
Comment (RonnyPfannschmidt) [reply .github/workflows/workflow.yaml:38]: i consider it a nice add as it takes the fetching out of the test run (

File: testing/conftest.py (hunk 5 above)
Comment (RonnyPfannschmidt) [reply testing/conftest.py]: fixed, thanks

File: testing/conftest.py (hunk 4 above)
Comment (RonnyPfannschmidt) [reply testing/conftest.py]: that was art of debug, removed again, thanks for your hint

File: testing/conftest.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply testing/conftest.py]: fixed, thanks

**Comment by quarckster** [comment]
@RonnyPfannschmidt it seems selenium standalone requires more than 10 seconds to up. Could you please to increase the timeout or switch back to `sudo`.

**Pull request merged**